php/php.dbgp/src/org/netbeans/modules/php/dbgp/breakpoints/ExceptionBreakpoint.java

```diff
@@ -27,6 +30,12 @@
  */
 public class ExceptionBreakpoint extends AbstractBreakpoint {
     private final String exceptionName;
+    @NullAllowed
+    private String message;
```

Comment:
What is the threading model of this class? I mean, is it safe to add these non-final fields here? In other words, is this class used only from one thread?

Reply:
This class is used in multiple threads. I added `synchronized` to the class methods that work with these fields. I hope I did it correctly.

Reply:
That seems to be "too much", IMHO.

The field `exceptionName` is final, so no problem. However, the `message` is not, so we need to take care of it. In this particular case, `volatile` (so `private String message;`) should be enough, since this is the only thing that can change.

I will add more comments to this class.